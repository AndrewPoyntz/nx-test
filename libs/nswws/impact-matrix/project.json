{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/nswws/impact-matrix/src",
  "projectType": "library",
  "generators": {
    "@nxext/stencil:component": {
      "style": "scss"
    }
  },
  "tags": [],
  "targets": {
    "test": {
      "executor": "@nxext/stencil:test",
      "outputs": ["{options.outputPath}"],
      "options": {
        "projectType": "library",
        "tsConfig": "libs/nswws/impact-matrix/tsconfig.lib.json",
        "configPath": "libs/nswws/impact-matrix/stencil.config.ts",
        "outputPath": "dist/libs/nswws/impact-matrix"
      }
    },
    "build": {
      "executor": "@nxext/stencil:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "projectType": "library",
        "tsConfig": "libs/nswws/impact-matrix/tsconfig.lib.json",
        "configPath": "libs/nswws/impact-matrix/stencil.config.ts",
        "outputPath": "dist/libs/nswws/impact-matrix"
      },
      "configurations": {
        "production": {
          "dev": false,
          "prod": true
        }
      }
    },
    "serve": {
      "executor": "@nxext/stencil:serve",
      "outputs": ["{options.outputPath}"],
      "options": {
        "projectType": "library",
        "tsConfig": "libs/nswws/impact-matrix/tsconfig.lib.json",
        "configPath": "libs/nswws/impact-matrix/stencil.config.ts",
        "outputPath": "dist/libs/nswws/impact-matrix"
      }
    },
    "e2e": {
      "executor": "@nxext/stencil:e2e",
      "outputs": ["{options.outputPath}"],
      "options": {
        "projectType": "library",
        "tsConfig": "libs/nswws/impact-matrix/tsconfig.lib.json",
        "configPath": "libs/nswws/impact-matrix/stencil.config.ts",
        "outputPath": "dist/libs/nswws/impact-matrix"
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": "libs/nswws/impact-matrix/**/*.{ts,tsx}"
      }
    }
  }
}
