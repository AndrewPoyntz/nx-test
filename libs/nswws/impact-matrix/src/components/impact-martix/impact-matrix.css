:host {
  display: block;
}

.impactMatrix {
  width: 100%;
  height: 100%;
}
table {
  width: inherit;
  height: inherit;
  border-spacing: 1px;
}
.likelihoodArrow {
  width: 20%;
  vertical-align: top;
}
.likelihoodArrow img {
  height: 80%;
  width: 85%;
}

.impactArrow img {
  width: 85%;
}
.cell {
  border: 1px solid #a1a0aaff;
  width: 20%;
  height: 100%;
  text-align: center;
  padding: 10% 1%;
  display: inline-block;
  margin: 0 0.1%;
  font-weight: bold;
  position: relative;
}
.green {
  background-color: #a3d700;
}
.yellow {
  background-color: #ffe923;
}
.amber {
  background-color: #ff9900;
}
.red {
  background-color: #cc0033;
  color: white;
}
.cell img {
  position: absolute;
  width: 50%;
  height: 50%;
  top: 25%;
  left: 25%;
}
