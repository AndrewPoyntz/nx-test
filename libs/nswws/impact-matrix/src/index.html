<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0"
    />
    <title>Stencil Component Starter</title>

    <script type="module" src="/build/impact-matrix.esm.js"></script>
    <script nomodule src="/build/impact-matrix.js"></script>
  </head>
  <body>
    <div style="width: 20em">
      <impact-matrix impact="2" likelihood="4"></impact-matrix>
    </div>
  </body>
</html>
