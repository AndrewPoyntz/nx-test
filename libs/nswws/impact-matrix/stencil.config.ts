import { Config } from '@stencil/core';

import { sass } from '@stencil/sass';

const angularValueAccessorBindings: ValueAccessorConfig[] = [];

import {
  angularOutputTarget,
  ValueAccessorConfig,
} from '@stencil/angular-output-target';

export const config: Config = {
  namespace: 'nswws-imact-matrix',
  taskQueue: 'async',
  plugins: [sass()],
  outputTargets: [
    {
      type: 'dist',
      esmLoaderPath: './loader',
    },
    {
      type: 'dist-custom-elements',
    },
    {
      type: 'docs-readme',
    },
    {
      type: 'www',
      serviceWorker: null, // disable service workers
    },

    angularOutputTarget({
      componentCorePackage: '@nx-test/nswws-impact-matrix',
      directivesProxyFile:
        '../../../../libs/nswws-impact-matrix-angular/src/generated/directives/proxies.ts',
      directivesArrayFile:
        '../../../../libs/nswws-impact-matrix-angular/src/generated/directives/index.ts',
      valueAccessorConfigs: angularValueAccessorBindings,
    }),
  ],
};
