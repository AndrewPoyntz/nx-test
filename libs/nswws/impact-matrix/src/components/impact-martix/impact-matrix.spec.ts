import { newSpecPage } from '@stencil/core/testing';
import { ImpactMatrix } from './impact-matrix.component';

describe('my-component', () => {
  it('renders', async () => {
    const { root } = await newSpecPage({
      components: [ImpactMatrix],
      html: '<impact-matrix></impact-matrix>',
    });
    expect(root).toMatchSnapshot();
  });

  it('renders with values', async () => {
    const { root } = await newSpecPage({
      components: [ImpactMatrix],
      html: `<impact-matrix impact="2" likelihood="4"></impact-matrix>`,
    });
    expect(root).toMatchSnapshot();
  });
});
