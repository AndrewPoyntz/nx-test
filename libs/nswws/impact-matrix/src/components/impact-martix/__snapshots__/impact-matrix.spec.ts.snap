// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-component renders 1`] = `
<impact-matrix>
  <mock:shadow-root>
    <div class="impactMatrix">
      <table>
        <tr>
          <td class="likelihoodArrow" rowspan="5">
            <img src="/assets/likelihood_arrow.svg">
          </td>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="amber cell"></td>
          <td class="cell red"></td>
        </tr>
        <tr>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="amber cell"></td>
          <td class="amber cell"></td>
        </tr>
        <tr>
          <td class="cell green"></td>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="amber cell"></td>
        </tr>
        <tr>
          <td class="cell green"></td>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="cell yellow"></td>
        </tr>
        <tr>
          <td class="impactArrow" colspan="4">
            <img src="/assets/impact_arrow.svg">
          </td>
        </tr>
      </table>
    </div>
  </mock:shadow-root>
</impact-matrix>
`;

exports[`my-component renders with values 1`] = `
<impact-matrix impact="2" likelihood="4">
  <mock:shadow-root>
    <div class="impactMatrix">
      <table>
        <tr>
          <td class="likelihoodArrow" rowspan="5">
            <img src="/assets/likelihood_arrow.svg">
          </td>
          <td class="cell green"></td>
          <td class="cell yellow">
            <img src="/assets/tick.svg">
          </td>
          <td class="amber cell"></td>
          <td class="cell red"></td>
        </tr>
        <tr>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="amber cell"></td>
          <td class="amber cell"></td>
        </tr>
        <tr>
          <td class="cell green"></td>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="amber cell"></td>
        </tr>
        <tr>
          <td class="cell green"></td>
          <td class="cell green"></td>
          <td class="cell yellow"></td>
          <td class="cell yellow"></td>
        </tr>
        <tr>
          <td class="impactArrow" colspan="4">
            <img src="/assets/impact_arrow.svg">
          </td>
        </tr>
      </table>
    </div>
  </mock:shadow-root>
</impact-matrix>
`;
